txtorcon/torconfig.py

```diff
@@ -1224,11 +1224,28 @@ def _do_setup(self, data):
         # can't just return in @inlineCallbacks-decorated methods
         defer.returnValue(self)
 
+    def _add_hidden_service(self, directory, ports, auth, ver, group_read, hs, directories):
+        if directory is not None:
```

Comment:
I would maybe move this to be an inner-function in _setup_hidden_services, and rename to maybe_add_hidden_service() to match other things in txtorcon since it's really *just* a helper for _setup_hidden_services. Also then you don't need to pass the 'directories' list into the method.